internal/provider/jira_product_configuration_resource.go

```diff
@@ -0,0 +1,443 @@
+
+func (t jiraProductConfigurationResourceType) GetSchema(ctx context.Context) (tfsdk.Schema, diag.Diagnostics) {
+	return tfsdk.Schema{
+		// This description is used by the documentation generator and the language server.
+		MarkdownDescription: "A Jira Product Configuration is the connection between a Product and a Jira Instance. It defines the Product's settings for pushing Findings to Jira.",
+
+		Attributes: map[string]tfsdk.Attribute{
+			"project_key": {
+				MarkdownDescription: "The Jira Project Key",
+				Optional:            true,
+				Type:                types.StringType,
+				Computed:            true,
+				PlanModifiers: tfsdk.AttributePlanModifiers{
+					stringDefault(""),
+				},
+			},
+
+			"issue_template_dir": {
+				MarkdownDescription: "The folder containing Django templates used to render the JIRA issue description. Leave empty to use the default jira_full templates.",
+				Optional:            true,
+				Type:                types.StringType,
+				Computed:            true,
+				PlanModifiers: tfsdk.AttributePlanModifiers{
+					stringDefault(""),
+				},
+			},
+
+			"push_all_issues": {
+				MarkdownDescription: "Automatically maintain parity with JIRA. Always create and update JIRA tickets for findings in this Product.",
+				Optional:            true,
+				Type:                types.BoolType,
+				Computed:            true,
+				PlanModifiers: tfsdk.AttributePlanModifiers{
+					boolDefault(false),
+				},
+			},
+
+			"enable_engagement_epic_mapping": {
+				MarkdownDescription: "Whether to map engagements to epics in Jira",
+				Optional:            true,
+				Type:                types.BoolType,
+				Computed:            true,
+				PlanModifiers: tfsdk.AttributePlanModifiers{
+					boolDefault(false),
+				},
+			},
+
+			"push_notes": {
+				MarkdownDescription: "Whether to push notes to Jira",
+				Optional:            true,
+				Type:                types.BoolType,
+				Computed:            true,
+				PlanModifiers: tfsdk.AttributePlanModifiers{
+					boolDefault(false),
+				},
+			},
+
+			"product_jira_sla_notification": {
+				MarkdownDescription: "Send SLA notifications as comments",
+				Optional:            true,
+				Type:                types.BoolType,
+				Computed:            true,
+				PlanModifiers: tfsdk.AttributePlanModifiers{
+					boolDefault(false),
+				},
+			},
+
+			"risk_acceptance_expiration_notification": {
+				MarkdownDescription: "Send Risk Acceptance expiration notifications as comments",
+				Optional:            true,
+				Type:                types.BoolType,
+				Computed:            true,
+				PlanModifiers: tfsdk.AttributePlanModifiers{
+					boolDefault(false),
+				},
+			},
+
+			"jira_instance_id": {
+				MarkdownDescription: "The ID of the Jira Instance to use for this Product",
+				Optional:            true,
+				Type:                types.StringType,
+			},
+
+			"product_id": {
+				MarkdownDescription: "The ID of the Product to configure",
```

Comment:
Can we put in an explanation here saying that it's "optional" but kind of required since either `product_id` or `engagement_id` must be defined?

```suggestion
				MarkdownDescription: "The ID of the Product to configure. Although optional, either the Product ID or the Engagement ID must be defined to create a Jira Product Configuration.",
```